scipy/optimize/_hessian_update_strategy.py

```diff
@@ -190,7 +190,7 @@ def update(self, delta_x, delta_grad):
             if self.init_scale == "auto":
                 scale = self._auto_scale(delta_x, delta_grad)
             else:
-                scale = float(self.init_scale)
+                scale = self.init_scale
```

Comment:
The CI failure looks related--may be good to update the PR/description to clarify that the change is extending beyond documentation only now.